torch/csrc/autograd/profiler_python.cpp

```diff
@@ -249,6 +256,28 @@ class ValueCache {
 
   void trimPrefixes();
 
+  template <CallType C>
+  typename Config<C>::cls_t set_class(
+      typename Config<C>::cache_t& cache,
+      const typename Config<C>::key_t& key,
+      const typename Config<C>::ephemeral_t& frame,
+      const PyCodeObject* code_match) {
```

Comment:
If we template get code on CallType then we can make this more generic. E.g. `TORCH_INTERNAL_ASSERT(code.get() == getCode<C>());`

Reply:
updated